cardano-cli/src/Cardano/CLI/EraBased/Script/Spend/Read.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.CLI.EraBased.Script.Spend.Read
+  ( CliSpendScriptWitnessError
+  , readSpendScriptWitness
+  )
+where
+
+import           Cardano.Api
+import           Cardano.Api.Shelley
+
+import           Cardano.CLI.EraBased.Script.Mint.Types (CliScriptWitnessError (..))
```

Comment:
The fact that you need to import `Mint` from `Spend` means that we should have a module for errors for common for all scripts.